File: minimum-time-visiting-all-points.ts
{
    const minTimeToVisitAllPoints = function(points) {
        let result = 0

        for(let i = 1; i < points.length; ++i) {
            const point_1 = points[i - 1]
            const point_2 = points[i]

            result += Math.max(Math.abs(point_1[0] - point_2[0]), Math.abs(point_1[1] - point_2[1]))
        }

        return result
    };

    console.log(minTimeToVisitAllPoints([[1,1],[3,4],[-1,0]]))
}

File: minimum-time-visiting-all-points.rs
pub fn min_time_to_visit_all_points(points: Vec<Vec<i32>>) -> i32 {
    let mut result = 0;

    for i in 1..points.len() {
        let point_1 = &points[i - 1];
        let point_2 = &points[i];

        let dx = (point_2[0] - point_1[0]).abs();
        let dy = (point_2[1] - point_1[1]).abs();

        result += dx.max(dy);
    }

    result
}
